examples/cli-chat/src/backend/cli_chat/core.clj

```diff
@@ -10,41 +10,50 @@
     [framework.route.core :as routes]
     [framework.session.core :as session]
     [framework.webserver.core :as ws]
-    [reitit.ring :as ring]))
+    [piotr-yuxuan.closeable-map :refer [closeable-map]]
+    [reitit.ring :as ring]
+    [xiana.commons :refer [rename-key]]))
 
 (def routes
   [["/" {:action index/handle-index}]
    ["/re-frame" {:action re-frame/handle-index}]
    ["/assets/*" (ring/create-resource-handler {:path "/"})]
    ["/chat" {:ws-action chat-action}]])
 
-(defn system
-  [config]
-  (let [deps {:routes                  (routes/reset routes)
-              :db                      (db/start (:framework.db.storage/postgresql config))
-              :webserver               (:framework.app/web-server config)
-              :role-set                (rbac/init (:framework.app/role-set config))
-              :auth                    (:framework.app/auth config)
-              :session-backend         (session/init-in-memory)
-              :router-interceptors     []
-              :web-socket-interceptors [interceptors/params
-                                        session/guest-session-interceptor]
-              :controller-interceptors [(interceptors/muuntaja)
-                                        interceptors/params
-                                        session/guest-session-interceptor
-                                        interceptors/view
-                                        interceptors/side-effect
-                                        db/db-access
-                                        rbac/interceptor]}]
-    (assoc deps :webserver (ws/start deps))))
+(defn ->system
+  [app-cfg]
+  (-> (config/config)
+      (merge app-cfg)
+      routes/reset
+      db/start
+      db/migrate!
+      rbac/init
+      (rename-key :framework.app/auth :auth)
+      session/init-in-memory
+      ws/start
+      (select-keys [:routes
+                    :db
+                    :rbac
+                    :auth
+                    :session-backend
+                    :webserver])
+      closeable-map))
```

Comment:
doesn't need to be closeable-map this one.

Reply:
It should be here once, or in two other places: in fixture, and in dev namespace...